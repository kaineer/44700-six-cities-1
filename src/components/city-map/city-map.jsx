import React, {Component} from 'react';
import {arrayOf, shape} from 'prop-types';
import {Place} from '../place/place';
import leaflet from 'leaflet';

export class CityMap extends Component {
  componentDidMount() {
    try {
      this._attachLeafletMap();
    } catch (err) {
      // Не даём упасть, если лифлет не подключился
      // console.log(err.stack.toString());
    }
  }

  render() {
    return (
      <section className="cities__map map" id="map"></section>
    );
  }

  _attachLeafletMap() {
    const city = [52.38333, 4.9];

    const icon = leaflet.icon({
      iconUrl: `img/pin1.svg`,
      iconSize: [30, 30]
    });

    const zoom = 12;
    const map = leaflet.map(`map`, {
      center: city,
      zoom,
      zoomControl: false,
      marker: true
    });
    map.setView(city, zoom);

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    this.props.places.forEach(({location}) => {
      leaflet.
        marker(location, {icon}).
        addTo(map);
    });
  }
}

CityMap.propTypes = {
  places: arrayOf(shape(Place.propTypes))
};
