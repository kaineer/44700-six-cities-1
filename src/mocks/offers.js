// src/mocks/offers.js

module.exports = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    rating: 93,
    isPremium: true,
    picture: `img/apartment-01.jpg`,
    type: `Apartment`,
  },
  {
    title: `Wood and stone place`,
    price: 60,
    rating: 20,
    isPremium: false,
    picture: `img/apartment-02.jpg`,
    type: `Place`,
  },
  {
    title: `Strange and abandoned grove`,
    price: 70,
    rating: 30,
    isPremium: false,
    picture: `img/apartment-03.jpg`,
    type: `Environment`,
  },
  {
    title: `Winter stronghold, just for you`,
    price: 300,
    rating: 99,
    isPremium: true,
    picture: `img/apartment-01.jpg`,
    type: `Castle`,
  }
];
