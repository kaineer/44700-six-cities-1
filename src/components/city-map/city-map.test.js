import React from 'react';
import {CityMap} from './city-map';
import {create} from 'react-test-renderer';

import offers from '../../mocks/offers';

describe(`CityMap`, () => {
  it(`should render correctly`, () => {
    const tree = create(
        <CityMap
          places={offers}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
