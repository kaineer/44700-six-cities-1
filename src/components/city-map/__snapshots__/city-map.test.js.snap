// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityMap should render correctly 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
